muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/notes/NotesRESTService.java

```diff
@@ -0,0 +1,151 @@
+package fi.otavanopisto.muikku.plugins.notes;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.ejb.Stateful;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import fi.otavanopisto.muikku.model.users.UserEntity;
+import fi.otavanopisto.muikku.plugin.PluginRESTService;
+import fi.otavanopisto.muikku.plugins.notes.model.Note;
+import fi.otavanopisto.muikku.schooldata.RestCatchSchoolDataExceptions;
+import fi.otavanopisto.muikku.session.SessionController;
+import fi.otavanopisto.muikku.users.UserEntityController;
+import fi.otavanopisto.security.rest.RESTPermit;
+import fi.otavanopisto.security.rest.RESTPermit.Handling;
+
+@RequestScoped
+@Stateful
+@Produces("application/json")
+@Path ("/notes")
+@RestCatchSchoolDataExceptions
+public class NotesRESTService extends PluginRESTService {
+
+  private static final long serialVersionUID = 6610657511716011677L;
+  
+  @Inject
+  private UserEntityController userEntityController;
+  
+  @Inject
+  private SessionController sessionController;
+  
+  @Inject
+  private NotesController notesController;
+  
+
+  // mApi() call (mApi().notes.note.create(noteRestModel)
+  //
+  // noteRestModel: = {
+  //  title: String, 
+  //  description: String, 
+  //  type: emun MANUAL
+  //  priority: enum LOW/NORMAL/HIGH, 
+  //  pinned: Boolean, 
+  //  owner: userIdentifier
+  //  };
+  @POST
+  @Path("/note")
+  @RESTPermit (handling = Handling.INLINE, requireLoggedIn = true)
+  public Response createNote(NoteRestModel note) {
+    
+      Note newNote = notesController.createNote(note.getTitle(), note.getDescription(), note.getType(), note.getPriority(), note.getPinned(), note.getOwner());
```

Comment:
Maybe check that if we're a student, we're only allowed to create notes for ourselves?